soh/src/code/z_actor.c

```diff
@@ -4983,15 +4983,27 @@ void Flags_UnsetEventInf(s32 flag) {
  * Tests if "randomizerInf" flag is set.
  */
 s32 Flags_GetRandomizerInf(RandomizerInf flag) {
-    return gSaveContext.randomizerInf[flag >> 4] & (1 << (flag & 0xF));
+    if (!IS_RANDO) {
+        //LUSLOG_ERROR("Tried to get randomizerInf flag \"%d\" outside of rando", flag);
+        //assert(false);
+        return 0;
+    }
+
+    return gSaveContext.ship.randomizerInf[flag >> 4] & (1 << (flag & 0xF));
 }
 
 /**
  * Sets "randomizerInf" flag.
  */
 void Flags_SetRandomizerInf(RandomizerInf flag) {
+    if (!IS_RANDO) {
+        LUSLOG_ERROR("Tried to set randomizerInf flag \"%d\" outside of rando", flag);
+        assert(false);
+        return;
+    }
+
```

Comment:
I understand why this was added while you were already around this code, but I'd suggest to separate this kinda stuff into a different PR down the line.